src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -350,7 +357,8 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 	}
 
 	/* check that table is empty if that is required */
-	if (requireEmpty && !LocalTableEmpty(relationId))
+	*emptyTable = LocalTableEmpty(relationId);
```

Comment:
You could remove the `requireEmpty` flag and introduce an `EnsureLocalTableEmpty()` function for callers that require the table to empty.